make/src/rule.rs

```diff
@@ -151,13 +155,31 @@ impl Rule {
     }
 
     /// A helper function to initialize env vars for shell commands.
-    fn init_env(&self, command: &mut Command, variables: &[VariableDefinition]) {
-        command.envs(variables.iter().map(|v| {
-            (
-                v.name().unwrap_or_default(),
-                v.raw_value().unwrap_or_default(),
-            )
-        }));
+    fn init_env(&self, env_macros: bool, command: &mut Command, variables: &[VariableDefinition]) {
+        let mut macros: Vec<(String, String)> = variables
+            .iter()
+            .map(|v| {
+                (
+                    v.name().unwrap_or_default(),
+                    v.raw_value().unwrap_or_default(),
+                )
+            })
+            .collect();
+
+        if env_macros {
+            // Retrieve environment variables from the system
+            let env_vars: Vec<(String, String)> = std::env::vars().collect();
+
+            // Filter and replace variables if they exist in the system environment
+            for (key, value) in env_vars {
+                // Remove any existing entry with the same key
+                macros.retain(|(k, _)| k != &key);
+                // Add the new environment variable
+                macros.push((key, value));
+            }
+        }
+
+        command.envs(macros);
     }
```

Comment:
This function can be rewritten in a more optimized way.
You can try to use `HashMap`s to not use `retain` which is O(n).